### ISSUE TITLE
oversized record error from tls

### ISSUE THREAD

**Issue by ketzacoatl** [hashicorp/vault#123]
I am testing out vault, and started off with TLS setup and enabled, but disabled for debugging the setup. My config looks like:

```
backend "file" {
    "path" = "/home/vault/data"
}

listener "tcp" {
    tls_disable = 1
#   tls_cert_file = "/home/vault/server.crt"
#   tls_key_file  = "/home/vault/server.key"
}
```

Vault starts and appears to be happy, with TLS disabled:

```
==> Vault server configuration:

         Log Level: info
           Backend: file
        Listener 1: tcp (addr: "127.0.0.1:8200", tls: "disabled")

==> Vault server started! Log data will stream in below:
```

But when I go to use the `vault` client in any way, I get this funky error from tls..

``` 
Mount error: Post https://127.0.0.1:8200/v1/sys/mounts/aws: tls: oversized record received with length 20527
```

or

```
# vault write secret/password value=itsasecret
Error writing data to secret/password: Put https://127.0.0.1:8200/v1/secret/password: tls: oversized record received with length 20527
```

**Comment by mitchellh** [comment]
You disabled TLS but the client will always use TLS by default. You must set `VAULT_ADDR` to `http://127.0.0.1:8200`.

I wonder how we can document this better? Or maybe we should just do a strings contains check on the error for "tls" :-\

**Comment by ketzacoatl** [comment]
hrm.. maybe the client could look to a config file in one of a few locations on the filesystem, with its runtime config?

FWIW, I am also confused by the `insecure` flag.. how does this relate? eg, if the addr isn't using https, how does using `insecure` change the behavior of vault?

```
  -insecure               Do not verify TLS certificate. This is highly
                          not recommended. This is especially not recommended
                          for unsealing a vault.
```

and how is it you are like me and never sleep?  : P


-J

**Comment by mitchellh** [comment]
Ah, `-insecure` disables certificat verification... "insecure SSL mode", rather than "no SSL mode". Sorry for the confusion.

**Comment by ketzacoatl** [comment]
Oh, in regards to documenting.. a more clear message would include something to the effect: _TLS is not configured correctly - are you trying to connect with a client over TLS when it is disabled on the server?_

**Comment by abaird** [comment]
I got this error too. When you are following the getting started guide on page http://www.vaultproject.io/intro/getting-started/deploy.html and you start the server it takes over your window so I switched to another one. In the next window, I continued entering the `vault init` command. This gives the error:
```
Error initializing Vault: Put https://127.0.0.1:8200/v1/sys/init: tls: oversized record received with length 20527
```
This is caused because in my new window, I don't have VAULT_ADDR set. To fix this, re-enter:
`export VAULT_ADDR='http://127.0.0.1:8200'`
